### PR TITLE
Fixing session sharing with IP matching.

### PR DESCRIPTION
**Pull request by bneradt**
This fixes session sharing using IP matching.

Note that this requires an updated version of microserver that supports keeping connections alive, hence the update to Pipefile to require the new version 1.0.4.

### CODE REVIEW

File: tests/gold_tests/pluginTest/cert_update/cert_update.test.py [hunk 1]
```diff
@@ -75,10 +75,6 @@
 # Set up plugin
 Test.PreparePlugin(Test.Variables.AtsExampleDir + '/plugins/c-api/cert_update/cert_update.cc', ts)
 
-ts.Disk.plugin_config.AddLine(
```
Comment (shinrich) [inline tests/gold_tests/pluginTest/cert_update/cert_update.test.py]: Probably want to pull these unrelated test cleanups to a separate PR.

**Comment by shinrich** [comment]
The core changes look good, and fixed a deployment blocker for us.

File: tests/gold_tests/pluginTest/cert_update/cert_update.test.py (hunk 1 above)
Comment (bneradt) [reply tests/gold_tests/pluginTest/cert_update/cert_update.test.py]: Sounds good. I reverted these other test changes from this PR.

Comment (bneradt) [reply tests/gold_tests/pluginTest/cert_update/cert_update.test.py]: Added a new PR for the test tweaks here:

https://github.com/apache/trafficserver/pull/6069

File: tests/gold_tests/session_sharing/session_management_client.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                        help="Port to use")
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG, format="%(message)s")
+
+    # Make a request on connection 1.
+    firstConnection = requests.Session()
+    firstConnection.get(
+        'http://127.0.0.1:{0}/one'.format(args.port),
+        headers={'Connection': "keep-alive",
```
Comment (bryancall) [inline tests/gold_tests/session_sharing/session_management_client.py]: 1. You shouldn't need `Connection: keep-alive` for HTTP/1.1.  Could test using both.
2. It looks like you are having the client close the connection. You can have keep-alive/close independently on the client without effecting the origin, and the reverse is true.

Comment (bneradt) [reply tests/gold_tests/session_sharing/session_management_client.py]: Thanks Bryan. This is helpful.

I removed the explicit keep-alives since they are not necessary, as you point out. microserver does the right thing and assumes a persistent connection.

I also simplified the test to simply use curl as the client since it can close the connection on the client side with each transaction and the server-side connection persists (and gets reused) as desired.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.
